app/ui/browser-australis/views/browser/decorations/statusbar.jsx

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import React, { Component, PropTypes } from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import { connect } from 'react-redux';
+
+import Style from '../../../../shared/style';
+
+import * as UIConstants from '../../../constants/ui';
+import * as UISelectors from '../../../selectors/ui';
+
+const STATUS_STYLE = Style.registerStyle({
+  position: 'fixed',
+  bottom: 0,
+  left: 0,
+  zIndex: UIConstants.STATUS_BAR_ZINDEX,
+  padding: '2px 8px',
+  backgroundColor: 'var(--theme-window-background)',
+  color: 'var(--theme-content-color)',
+});
+
+class StatusBar extends Component {
+  constructor(props) {
+    super(props);
+    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
```

Comment:
I think the latest decorators spec is quite different form what's in Babel:
https://github.com/wycats/javascript-decorators/blob/master/interop/reusability.md

On Tue, Sep 6, 2016 at 1:32 PM Victor Porof <notifications@github.com>
wrote:

> In app/ui/browser-australis/views/browser/decorations/statusbar.jsx
> <https://github.com/mozilla/tofino/pull/1054#discussion_r77624723>:
>
> > +import * as UISelectors from '../../../selectors/ui';
> > +
> > +const STATUS_STYLE = Style.registerStyle({
> > +  position: 'fixed',
> > +  bottom: 0,
> > +  left: 0,
> > +  zIndex: UIConstants.STATUS_BAR_ZINDEX,
> > +  padding: '2px 8px',
> > +  backgroundColor: 'var(--theme-window-background)',
> > +  color: 'var(--theme-content-color)',
> > +});
> > +
> > +class StatusBar extends Component {
> > +  constructor(props) {
> > +    super(props);
> > +    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
>
> ES7 decorators are a very elegant way of getting around this:
>
> @PureRenderclass StatusBar extends Component {
> }
>
> However, they were still experimental (disabled by default) in babel:
> https://babeljs.io/docs/plugins/syntax-decorators/
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/tofino/pull/1054/files/a6359baa2d712378c6b667f006a5b0584b34fde3#r77624723>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEp4iNY1O4onSA1o98M4Hgu-NRQfSXbks5qnV1HgaJpZM4J1LcF>
> .
>